packages/alfa-rules/src/sia-er62/rule.ts

```diff
@@ -349,8 +328,8 @@ namespace Distinguishable {
       // Things like text decoration and backgrounds risk blending with the
       // container element. We therefore need to check if these can be distinguished
       // from what the container element might itself set.
-      hasDistinguishableBackground(container, device, context),
       hasDistinguishableContrast(container, device, context),
+      hasDistinguishableBackground(container, device, context),
```

Comment:
Indeed, the `or` is lazy and escape as soon as one condition is `true`. Which can make the cache not being filled, and then creates problem in the expectation when trying to `.get` it.
By copying the cache filling in the expectation, we ensure the cache always get the correct data (and the cache implementation makes sure we do not compute the same thing twice and don't waste time). This should allow to keep the checks here in alphabetical order which is easier to navigate for humans 😄